shared/chat/conversation/list/container.js

```diff
@@ -0,0 +1,157 @@
+// @flow
+import * as Constants from '../../../constants/chat'
+import * as Creators from '../../../actions/chat/creators'
+import HiddenString from '../../../util/hidden-string'
+import ListComponent from '.'
+import {List, Map} from 'immutable'
+import {connect} from 'react-redux'
+import {downloadFilePath} from '../../../util/file'
+import {navigateAppend} from '../../../actions/route-tree'
+import {pick} from 'lodash'
+import {compose} from 'recompose'
+
+import type {OpenInFileUI} from '../../../constants/kbfs'
+import type {Options} from '../messages'
+import type {Props, OptionsFn} from '.'
+import type {OwnProps, StateProps, DispatchProps} from './container'
+import type {TypedState} from '../../../constants/reducer'
+
+const mapStateToProps = (state: TypedState, {editLastMessageCounter, listScrollDownCounter, onFocusInput}: OwnProps): StateProps => {
+  const selectedConversationIDKey = Constants.getSelectedConversation(state)
+  const you = state.config.username || ''
+  const origFollowingMap = state.config.following
+  const origMetaDataMap = state.chat.get('metaData')
+
+  let firstNewMessageID = null
+  let followingMap = Map()
+  let messages = List()
+  let metaDataMap = Map()
+  let moreToLoad = false
+  let participants = List()
+  let editingMessage = null
+  let supersedes = null
+  let supersededBy = null
+  let validated = false
+
+  if (selectedConversationIDKey && Constants.isPendingConversationIDKey(selectedConversationIDKey)) {
+    const tlfName = Constants.pendingConversationIDKeyToTlfName(selectedConversationIDKey)
+    if (tlfName) {
+      participants = List(tlfName.split(','))
+      followingMap = pick(origFollowingMap, participants.toArray())
+      metaDataMap = origMetaDataMap.filter((k, v) => participants.contains(v))
+      validated = true
+    }
+  } else if (selectedConversationIDKey && selectedConversationIDKey !== Constants.nothingSelected) {
+    const conversationState = state.chat.get('conversationStates').get(selectedConversationIDKey)
```

Comment:
yeah, once i have this stuff all connected, we can move to entities and conversationStates will go away